neqo-transport/src/connection/mod.rs

```diff
@@ -2625,7 +2629,7 @@ impl Connection {
             let reset_token = remote
                 .get_bytes(tparams::STATELESS_RESET_TOKEN)
                 .map_or_else(ConnectionIdEntry::random_srt, |token| {
-                    <[u8; 16]>::try_from(token).unwrap()
+                    <[u8; 16]>::try_from(token).unwrap_or_else(|_| ConnectionIdEntry::random_srt())
```

Comment:
This seems like a mistake.  Of anything, I would return `Err(Error::TransportErrorParameter)`.